addons/stock_account/tests/test_anglo_saxon_valuation_reconciliation.py

```diff
@@ -0,0 +1,159 @@
+            line_form.account_id = self.company_data['default_account_stock_out']
+            line_form.quantity = 1
+            line_form.tax_ids.clear()
+        invoice_3 = move_form.save()
+
+        invoice_3.action_post()
+
+        # Creating stock moves and associated sale order lines
+        stock_location = self.env['stock.warehouse'].search([
+            ('company_id', '=', self.env.company.id),
+        ], limit=1).lot_stock_id
+
+        out_picking = self.env['stock.picking'].create({
+            'location_id': stock_location.id,
+            'location_dest_id': self.ref('stock.stock_location_customers'),
+            'picking_type_id': stock_location.warehouse_id.out_type_id.id,
+        })
+
+        sm_1 = self.env['stock.move'].create({
+            'name': products[0].name,
+            'product_id': products[0].id,
+            'product_uom_qty': 1,
+            'product_uom': products[0].uom_id.id,
+            'location_id': out_picking.location_id.id,
+            'location_dest_id': out_picking.location_dest_id.id,
+            'picking_id': out_picking.id,
+            'account_move_ids': invoice_1 | invoice_3,
+            'state': 'done'
+        })
+
+        sale_order_line_1 = self.env['sale.order.line'].create({
+            'product_id': products[0].id,
+            'order_id': sale_order.id,
+            'move_ids': sm_1
+        })
+
+        sm_1.sale_line_id = sale_order_line_1.id
+
+        sm_2 = self.env['stock.move'].create({
+            'name': products[1].name,
+            'product_id': products[1].id,
+            'product_uom_qty': 1,
+            'product_uom': products[1].uom_id.id,
+            'location_id': out_picking.location_id.id,
+            'location_dest_id': out_picking.location_dest_id.id,
+            'picking_id': out_picking.id,
+            'account_move_ids': invoice_1 | invoice_2,
+            'state': 'done'
+        })
+
+        sale_order_line_2 = self.env['sale.order.line'].create({
+            'product_id': products[1].id,
+            'order_id': sale_order.id,
+            'move_ids': sm_2
+        })
+
+        sm_2.sale_line_id = sale_order_line_2.id
+
+        invoice_1.invoice_line_ids.sale_line_ids = sale_order_line_1 | sale_order_line_2
+
+        # Creating a stock valuation layer for invoice_2 to populate the no_exchange_reconcile_plan
+        svl_vals = {
+            'company_id': self.env.company.id,
+            'product_id': products[1].id,
+            'description': "description",
+            'unit_cost': products[1].standard_price,
+            'quantity': 1,
+        }
+
+        invoice_2.stock_valuation_layer_ids |= self.env['stock.valuation.layer'].create(svl_vals)
+        invoice_2.stock_valuation_layer_ids.stock_valuation_layer_id |= self.env['stock.valuation.layer'].create(svl_vals)
+
+        try:
+            invoice_1._post()
+        except UserError as e:
+            raise AssertionError(e)
```

Comment:
Simply post the invoice, then test the status 
```python
self.assertEqual(invoice_1.state, 'posted')
``` 
If the `_post` fail or the state in not `posted`, the test will fail.